src/signing.ts

```diff
@@ -0,0 +1,118 @@
+import { encodeDagJson, decodeDagJson } from './utils'
+import { createJWS, verifyJWS } from 'did-jwt'
+import base64url from 'base64url'
+import stringify from 'fast-json-stable-stringify'
+
+interface GeneralSignature {
+  header?: Object;
+  protected?: string;
+  signature: string;
+}
+
+interface GeneralJWS {
+  payload: string;
+  signatures: Array<GeneralSignature>;
+}
+
+interface DagSignature {
+  header?: Object;
+  protected?: Object;
+  signature: Buffer;
+}
+
+interface DagJWS {
+  payload: Object | Buffer;
+  signatures: Array<DagSignature>;
+}
+
+interface PublicKey {
+  publicKeyHex?: string;
+  publicKeyBase64?: string;
+}
+type Signer = (data: string) => string
```

Comment:
should this be `type Signer = (data: string) => Promise<EcdsaSignature | string>`?

Reply:
It's fine to have it be string since it's stricter than the signature you're suggesting. `EcdsaSignature` is also legacy in did-jwt.

Reply:
cool 👍

Reply:
hm, seems like this might be an issue when compiling. Will look closer at it!